src/groupByKey.ts

```diff
@@ -2,6 +2,14 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((groups, item) => {
+    const value = item[key];
+    const group = groups[value as string] || [];
```

Comment:
https://mateacademy.slack.com/archives/C04N8TU8PN2/p1680773288486249

Reply:
Your code looks good! But you should correct this little thing) check link in previous message)

Reply:
Also there are some other 'as' in your code, you should correct it too